File: Lab5/eshop-app/src/main/java/com/yube/services/ProductCategoryService.java
package com.yube.services;

import com.yube.model.dto.ProductCategory;
import com.yube.model.entity.ProductCategoryEntity;
import com.yube.model.mapping.ProductCategoryMapper;
import com.yube.model.mapping.utils.MappingUtils;
import com.yube.repositories.declaration.ProductCategoryRepository;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.BeanUtils;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
@AllArgsConstructor
@Slf4j
public class ProductCategoryService {

    private final ProductCategoryMapper productCategoryMapper;
    private final ProductCategoryRepository productCategoryRepository;

    public List<ProductCategory> getAll() {
        return MappingUtils.mapAll(productCategoryRepository.getAll(), productCategoryMapper::map);
    }

    public ProductCategory getById(String id) {
        return productCategoryMapper.map(productCategoryRepository.get(UUID.fromString(id)).get());
    }

    public ProductCategory getDefault() {
        return productCategoryMapper.map(productCategoryRepository.findFirstBy().orElse(null));
    }

    public ProductCategory create(ProductCategory productCategory) {
        ProductCategoryEntity productCategoryEntity = productCategoryMapper.unmap(productCategory);
        return productCategoryMapper.map(productCategoryRepository.create(productCategoryEntity));
    }

    public ProductCategory update(ProductCategory productCategory) {
        ProductCategoryEntity sourceProductCategoryEntity = productCategoryMapper.unmap(productCategory);
        ProductCategoryEntity targetProductCategoryEntity = productCategoryRepository.get(sourceProductCategoryEntity.getId()).get();
        BeanUtils.copyProperties(sourceProductCategoryEntity, targetProductCategoryEntity);
        ProductCategoryEntity updatedProductCategoryEntity = productCategoryRepository.update(targetProductCategoryEntity);
        return productCategoryMapper.map(updatedProductCategoryEntity);
    }

    public void delete(String id) {
        productCategoryRepository.delete(UUID.fromString(id));
    }
}

File: Lab6/eshop-ui-react/src/services/product-category.service.js
import http from "../http-common";
import {apiUrls} from "../api-urls";

class ProductCategoryService {

    basicUrl = apiUrls.API_PRODUCT_CATEGORIES;

    getAll() {
        return http.get(this.basicUrl);
    }

    getById(id) {
        const url = `${this.basicUrl}/${id}`;
        return http.get(url);
    }

    getDefault() {
        const url = `${this.basicUrl}/default`;
        return http.get(url);
    }

    create(productCategory) {
        return http.post(this.basicUrl, productCategory);
    }

    update(productCategory) {
        return http.put(this.basicUrl, productCategory);
    }

    delete(id) {
        const url = `${this.basicUrl}/${id}`;
        return http.delete(url);
    }
}

export default new ProductCategoryService();
